### DISCUSSION TITLE
Collections not showing up in DB in query app after creating it in loader app - using the Java SDK

### DISCUSSION THREAD

**Discussion by fgreco55** [milvus-io/milvus#27415] [Q&A and General discussion]
I have two Java programs.  One creates data and stores it in a Milvus DB.  That program can successfully query() the DB (after some great help from the support team).  That app works fine.

The second program queries the DB.  It can successfully open the connection with the associated DB, but when I list the collections in it, it is empty.  I'm a vector DB noob (but coming from RDBMS world).  I also tried to load the collection, but Milvus cannot find the collection.  Any suggestions?

**Comment by yhmo** [comment]
Firstly list the db:
```
        R<ListDatabasesResponse> ddd = milvusClient.listDatabases();
        System.out.print(ddd.getData());
```
If the first program is using a database named "XXX" to create collections, the collections are belong to the default database, and not visible for other database. If the second program is also using the database "XXX" to list collectons, it should can see the collections. Otherwise it is a bug.

**Reply by fgreco55** [reply]
That is exactly what I did in my 2nd app.  The database is there, but my collections are not.

The last line of my "loader" app gets the collection stats fine.  My "query" app can see my databases but not any collections.  Is it somehow looking at the default database and not mine?

**Reply by fgreco55** [reply]
To clarify what you are saying:
If I use a database named "XXX", that database is associated with the client connection.  When I create collections, those collections should belong in database "XXX".   They should not belong to the default database.   Any other app that uses database "XXX" should see my collections.   They do not.

I have a database "frankdb" and a collection "frankcollection".  I can query() this collection fine in the creation app.  But separate apps cannot see that collection.  They can see the database, but not the collection.  This seems like a severe bug.

2023-09-28 23:58:50 io.milvus.client.AbstractMilvusGrpcClient logInfo 
INFO: LoadCollectionParam(databaseName=null, collectionName=frankcollection, syncLoad=true, syncLoadWaitingInterval=500, syncLoadWaitingTimeout=60, replicaNumber=1, refresh=false)
***ERROR Cannot load collection:  collection frankdb:frankcollection: collection not found
2023-09-28 23:58:50 io.milvus.client.AbstractMilvusGrpcClient logInfo 
INFO: ShowCollectionsParam{collectionNames='[]', showType=All}
R{status=0, data=status {
}
}

**Comment by yhmo** [comment]
I just use Java sdk to test your case.
I have a database named "frankdb". The first program create a collection "AAA" in the "frankdb":
```
public static void main(String[] args) {
        MilvusServiceClient milvusClient = new MilvusServiceClient(ConnectParam.newBuilder()
                .withHost("localhost")
                .withPort(19530)
                .withDatabaseName("frankdb")
                .build());

        R<ListDatabasesResponse> r = milvusClient.listDatabases();
        System.out.println(r.getData());

        String field1Name = "id_field";
        String field2Name = "vector_field";
        FieldType field1 = FieldType.newBuilder()
                .withPrimaryKey(true)
                .withAutoID(true)
                .withDataType(DataType.Int64)
                .withName(field1Name)
                .build();

        FieldType field2 = FieldType.newBuilder()
                .withDataType(DataType.BinaryVector)
                .withName(field2Name)
                .withDimension(dimension)
                .build();

        // create collection
        CreateCollectionParam createParam = CreateCollectionParam.newBuilder()
                .withCollectionName("AAA")
                .addFieldType(field1)
                .addFieldType(field2)
                .build();

        R<RpcStatus> createR = milvusClient.createCollection(createParam);
        System.out.println(createR.getData());

        R<ListCollectionsResponse> lc = milvusClient.listCollections(ListCollectionsParam.newBuilder().build());
        System.out.println(lc.getData());
}
```

The collection is created successfully:
```
status {
}
db_names: "frankdb"
db_names: "default"

2023-09-30 23:53:52 io.milvus.client.AbstractMilvusGrpcClient logInfo 
INFO: CreateCollectionParam(collectionName=AAA, shardsNum=0, description=, fieldTypes=[FieldType{name='id_field', type='Int64', primaryKey=true, partitionKey=false, autoID=true, params={}}, FieldType{name='vector_field', type='BinaryVector', primaryKey=false, partitionKey=false, autoID=false, params={dim=128}}], partitionsNum=0, consistencyLevel=BOUNDED, databaseName=null, enableDynamicField=false)
RpcStatus{msg='Success'}
2023-09-30 23:53:52 io.milvus.client.AbstractMilvusGrpcClient logInfo 
INFO: ListCollectionsParam(showCollectionsParam=ShowCollectionsParam{collectionNames='[]', showType=All})
2023-09-30 23:53:52 io.milvus.client.AbstractMilvusGrpcClient logInfo 
INFO: ShowCollectionsParam{collectionNames='[]', showType=All}
ListCollectionsResponse(collectionNames=[AAA])

Process finished with exit code 0
```

Then I use the second program to list collections of the "frankdb":

 ```
public static void main(String[] args) {
        // Connect to Milvus server. Replace the "localhost" and port with your Milvus server address.
        MilvusServiceClient milvusClient = new MilvusServiceClient(ConnectParam.newBuilder()
                .withHost("localhost")
                .withPort(19530)
                .withDatabaseName("frankdb")
                .build());

        R<ListDatabasesResponse> r = milvusClient.listDatabases();
        System.out.println(r.getData());

        R<ListCollectionsResponse> lc = milvusClient.listCollections(ListCollectionsParam.newBuilder().build());
        System.out.println(lc.getData());
}
```

No problem:
```
status {
}
db_names: "default"
db_names: "frankdb"

2023-09-30 23:54:09 io.milvus.client.AbstractMilvusGrpcClient logInfo 
INFO: ListCollectionsParam(showCollectionsParam=ShowCollectionsParam{collectionNames='[]', showType=All})
2023-09-30 23:54:09 io.milvus.client.AbstractMilvusGrpcClient logInfo 
INFO: ShowCollectionsParam{collectionNames='[]', showType=All}
ListCollectionsResponse(collectionNames=[AAA])

Process finished with exit code 0
```
